optimizely/event/processor_test.go

```diff
@@ -83,23 +102,92 @@ func TestDefaultEventProcessor_ProcessBatch(t *testing.T) {
 	result, ok := (processor.EventDispatcher).(*MockDispatcher)
 
 	if ok {
-		assert.Equal(t, 1, len(result.Events))
-		evs := result.Events[0]
-		assert.Equal(t, 4, len(evs.Event.Visitors))
+		assert.Equal(t, 1, result.Events.Size())
+		evs := result.Events.Get(1)
+		logEvent, _ := evs[0].(LogEvent)
+		assert.Equal(t, 4, len(logEvent.Event.Visitors))
 	}
 }
 
-func TestBatchEventProcessor_FlushesOnClose(t *testing.T) {
+func TestDefaultEventProcessor_QSizeMet(t *testing.T) {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	processor := NewEventProcessor(exeCtx, ProcessorQueueSize(2), ProcessorFlushInterval(100), func (qp *QueueingEventProcessor) {
+		qp.Q = NewInMemoryQueue(2)
+		qp.EventDispatcher = &MockDispatcher{Events:NewInMemoryQueue(100)}
+	})
+
+	impression := BuildTestImpressionEvent()
+	conversion := BuildTestConversionEvent()
+
+	processor.ProcessEvent(impression)
+	processor.ProcessEvent(impression)
+
+	assert.Equal(t, 2, processor.EventsCount())
+
+	timers.Sleep(exeCtx.GetContext(), 100 * time.Millisecond)
```

Comment:
Is there a reason you need to use a 3rd party timer? I'd prefer if we can just keep it simple with out-of-the-box. The reason is that we'd have to document yet another library we use in the OSS tracker